### DISCUSSION TITLE
[keyball61] １行（ASDFG行）のキー処理がうまくいかない

### DISCUSSION THREAD

**Discussion by O-23K** [Yowkees/keyball#280] [Q&A]
『Keyball61』の組み立てで、ボール無し左手側ASDFG行のキー（Tabキー含む）の処理がうまくいきません。

事象としては以下のような挙動となります。

１．D、F 、Gキーは反応しない
２．Tabキー、A 、 Sは『そのキーより上部のキー』が一緒に入力される
　＞具体的には、Aだと1qa、Sだと2wsが一度に入力されます。ｚとｘは入りません。
　＞Tabキーは、Esc＋Del＋Tabが一度に入力されているように見えます。

Remapのキーテストで確認すると、１のキー群は初期段階で灰色で押せなくなっています。２のキー群は押すと所属する列全体が反応します。

ASDFG行以外のキーは問題なく認識されています。

テスターでダイオードの確認、目視ではんだが他の端子に絡んでいないかの確認などはしたのですが、今のところ、特におかしな点は見つけられていません。

正直なところ、今回が初自作キーボードでして、調査に関する引き出しも少なく途方に暮れておりまして……この辺をこんな感じで調べたらいいよ！　レベルでも構いませんので、何かしら教えていただけるとありがたいです。

**Comment by Yowkees** [comment]
ご連絡ありがとうございます。

症状を聞いた感じですと、ダイオードの向き間違いかテンプラかと思われますが、確認済ということですので、ProMicro裏の半田付けで隣とショートしている端子がないかご確認いただければと思います。
半田付け面のお写真をみせていただけますでしょうか？なにかアドバイスできるかも知れません。

お手数ですがご対応お願いいたします。

**Comment by O-23K** [comment]
以下が基盤の状態となります。

あわせて、テスターで確認した内容も併記しておきます（テスターそのものも、うまく使えているか不明なので、念のため試した内容をそのまま書きます）。

写真１
![IMG_1](https://github.com/Yowkees/keyball/assets/140047650/e39ba544-efd6-4417-a42f-96e67d787b2c)

・写真１の状態にして、ダイオードの左側にテスターの赤、右側に黒をつなげて、抵抗が表示されるのを確認（赤と黒を逆にすると無反応）。

・写真１の状態にして、赤線から『左サイド』のキーソケットの左側にテスターの赤、右側に黒をつなげて、抵抗が表示されるのを確認（赤と黒を逆にすると無反応）。

・写真１の状態にして、赤線から『右サイド』のキーソケットの左側にテスターの黒、右側に赤をつなげて、抵抗が表示されるのを確認（赤と黒を逆にすると無反応）。赤線の左サイドとは赤と黒を逆にしないと反応しませんでした。

・写真１の状態にして親指位置のキーソケットは、ノーマルプロファイルの左側にテスターの黒、右側に赤をつなげて、抵抗が表示されるのを確認。ロープロファイルも同様。

写真２
![IMG_2](https://github.com/Yowkees/keyball/assets/140047650/3c61cbc0-2ab3-4c33-a64c-977bebdaca85)

写真３
![IMG_3](https://github.com/Yowkees/keyball/assets/140047650/753b14ed-6909-443d-a1a0-03e4aba5d5f8)

・Gの位置にある（LED３と４の間にある）キーソケットの左側にテスターの赤、テスターの黒をProMicroの裏面（写真２）、ProMicroの表面（写真３）のはんだ付けした各端子につなげて、抵抗が表示されるのを確認。
